src/main/java/com/autotune/analyzer/application/Tunable.java

```diff
@@ -63,6 +64,22 @@
     */
 
 	public ArrayList<String> sloClassList;
+	public List<String> categoricalValueList;
+
+	public Tunable(String name, double step, String valueType, Map<String, String> queries, ArrayList<String> sloClassList, String layerName, List<String> categoricalValueList) {
+
+		this.queries = queries;
+		this.name = Objects.requireNonNull(name, TUNABLE_NAME_EMPTY);
+		this.valueType = Objects.requireNonNull(valueType, VALUE_TYPE_NULL);
+		this.sloClassList = Objects.requireNonNull(sloClassList, INVALID_SLO_CLASS);
+		this.step = Objects.requireNonNull(step, ZERO_STEP);
+		if (AutotuneSupportedTypes.LAYERS_SUPPORTED.contains(layerName))
+			this.layerName = layerName;
+		else
+			this.layerName = "generic";
```

Comment:
1. Please move common code between the two constructors to a separate function
2. Add a javadoc for both the constructors to indicate the difference in the two types
3. Use the same format as the other constructor, have all the parameters on separate lines